public/components/homepage.js

```diff
@@ -6,32 +6,32 @@ let locationsTemp = [
   "Mendrisio"
 ];
 
+
 // Container for the search bar and the search button
 class SearchContainer extends React.Component {
   searchRequest(e) {
     // Function triggers only on button click (!e.keyCode) or enter key pressed
     if (!e.keyCode || e.keyCode == 13) {
       let searchWarning = document.getElementById('search-warning');
-      let searchName = document.getElementById('search-name').value;
-      
+      let searchName = document.getElementById('search-what').value;
       if (searchName.match(/[^A-Za-z ]/)) {
-        searchWarning.innerHTML = "Only alphabetic characters allowed.";
+        searchWarning.innerHTML = 'Only alphabetic characters allowed.';
```

Comment:
We should allow also numbers and some symbols like `-` and `+`. Example: searching by phone number, or `"HTML5"`, ...